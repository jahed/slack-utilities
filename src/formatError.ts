/**
 * Format an error message to output to Slack.
 */
function formatError(error: Error): string {
    return `${error}\n\`\`\`\n${error.stack}\n\`\`\``
}

export { formatError }
