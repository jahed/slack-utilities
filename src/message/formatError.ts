/**
 * Format an error message to output to Slack.
 */
export const formatError = (error: Error): string => {
  return `${error}\n\`\`\`\n${error.stack}\n\`\`\``
}
