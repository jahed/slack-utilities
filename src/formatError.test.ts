import { formatError } from './formatError'

test('should format error', () => {
    const result = formatError(new Error('This is the message'))
    expect(result).toContain('This is the message')
})
