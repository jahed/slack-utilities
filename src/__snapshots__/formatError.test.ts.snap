// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format error 1`] = `
"Error: This is the message
\`\`\`
Error: This is the message
    at Object.<anonymous> (C:\\\\Users\\\\Jahed\\\\projects\\\\slack\\\\src\\\\formatError.test.ts:4:32)
    at Object.asyncFn (C:\\\\Users\\\\Jahed\\\\projects\\\\slack\\\\node_modules\\\\jest-jasmine2\\\\build\\\\jasmine_async.js:124:345)
    at e (C:\\\\Users\\\\Jahed\\\\projects\\\\slack\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queue_runner.js:46:12)
    at mapper (C:\\\\Users\\\\Jahed\\\\projects\\\\slack\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queue_runner.js:34:499)
    at promise.then (C:\\\\Users\\\\Jahed\\\\projects\\\\slack\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queue_runner.js:74:39)
    at process._tickCallback (internal/process/next_tick.js:109:7)
\`\`\`"
`;
